app/services/metrics/merge_time/per_user_project.rb

```diff
@@ -1,42 +1,38 @@
 module Metrics
   module MergeTime
     class PerUserProject < Metrics::BaseDevelopmentMetrics
-      BATCH_SIZE = 500
-
       private
 
       def process
         ActiveRecord::Base.transaction do
-          entities = Hash.new(0)
-          pull_requests_in_batches do |pull_request|
-            entity = find_user_project(pull_request.owner, pull_request.project)
-            entities[entity] += 1
-            merge_time = calculate_merge_time(pull_request)
-
+          merge_times_in_batches do |project_id, owner_id, merge_time_value|
+            entity = find_user_project(owner_id, project_id)
             create_or_update_metric(entity.id, UsersProject.name,
-                                    metric_interval, merge_time, :merge_time)
+                                    merge_time_interval, merge_time_value, :merge_time)
           end
-          calculate_metrics_avg(entities, :merge_time)
         end
       end
 
-      def pull_requests_in_batches
-        filtered_pull_requests.find_each(batch_size: BATCH_SIZE).lazy.each do |pull_request|
-          yield(pull_request)
+      def merge_times_in_batches
+        filtered_merge_times.each do |project_id, owner_id, merge_time_value|
```

Comment:
Why did you stop using BATCH_SIZE and `lazy`?

Reply:
I don't have an answer for that haha
@hvilloria do you? Otherwise I'll just put them back

Reply:
Sorry, I do have an answer.
`filtered_merge_times` no longer returns an `ActiveRecord_Relation`, it returns an `Array`. So we can't use `find_each` anymore.
Given this, I'll just delete this `merge_times_in_batches` method, as it's useless now